uaclient/gpg.py

```diff
@@ -0,0 +1,33 @@
+import logging
+from uaclient import exceptions
+from uaclient import util
+
+
+def export_gpg_key_from_keyring(
+        key_id: str, source_keyring_file: str,
+        destination_keyfile: str) -> None:
+    """Export a specific key from source_keyring_file into destination_keyfile
+
+    :param key_id: Long fingerprint of key to export.
+    :param source_keyring_file: The keyring file from which to export.
+    :param destination_keyfile: The filename created with the single exported
+        key.
+
+    :raise KeyError: If specified key does not exist in the source_keyring_file
+    :raise UserFacingError: Any other GPG errors
+    """
+    export_cmd = ['gpg', '--no-auto-check-trustdb', '--options', '/dev/null',
+                  '--no-default-keyring', '--keyring', source_keyring_file,
+                  '--export', key_id]
+    logging.debug('Exporting gpg key %s from %s', key_id, source_keyring_file)
+    try:
+        out, _err = util.subp(export_cmd, decode=False)
+    except util.ProcessExecutionError as exc:
+        with util.disable_log_to_console():
+            logging.error(str(exc))
+        raise exceptions.UserFacingError(
+            "Unable to export gpg keys from keyring %s" % source_keyring_file)
+    if not out:
+        raise KeyError("gpg key '%s' not found in %s" % (
```

Comment:
We should not be using `KeyError` for anything other than dictionary keys.

Reply:
roger, using UserFacingError.